File: 5/2.py
with open("input", "r") as file:
    text = file.read().splitlines()

def decode_seat(seat: str):
    lower = 0
    upper = 127
    left = 0
    right = 7

    ans1, ans2 = 0, 0

    for i, letter in enumerate(seat):
        if letter == 'F':
            upper = lower + (upper - lower)//2
            ans1 = upper
        elif letter == 'B':
            lower = lower + (upper - lower)//2 + 1
            ans1 = lower
        
        elif letter == 'L':
            right = left + (right - left)//2
            ans2 = right
        elif letter == 'R':
            left = left + (right - left)//2 + 1
            ans2 = left
        
    return ans1*8+ans2

ids = sorted([decode_seat(i) for i in text])
for i, j in enumerate(ids[1:]):
    if j - ids[i] == 2:
        print((j + ids[i])//2)

File: 2020/5/2.rs
use std::fs::read_to_string;

fn decode_seat(seat: &str) -> i32 {
    let mut lower = 0;
    let mut upper = 127;
    let mut left = 0;
    let mut right = 7;

    let mut ans1 = 0;
    let mut ans2 = 0;

    for letter in seat.chars() {
        if letter == 'F' {
            upper = lower + (upper - lower) / 2;
            ans1 = upper;
        }else if letter == 'B' {
            lower = lower + (upper - lower) / 2 + 1;
            ans1 = lower
        }else if letter == 'L'{
            right = left + (right - left) / 2;
            ans2 = right;
        } else {
            left = left + (right - left ) / 2 + 1;
            ans2 = left;
        }
    }
    
    ans1*8 + ans2
}

fn main() {
    let input = read_to_string("./input").unwrap();

    let mut ids: Vec<i32> = input.lines().map(decode_seat).collect();
    ids.sort();

    for (i, j) in ids[1..].iter().enumerate() {
        if j - ids[i] == 2 {
            println!("{}", (j + ids[i]) / 2);
        }
    }
}